server.js

```diff
@@ -1,32 +1,150 @@
 import express from 'express'
 import bodyParser from 'body-parser'
 import cors from 'cors'
+import booksData from './data/books.json'
 
-// If you're using one of our datasets, uncomment the appropriate import below
-// to get started!
-// 
-// import goldenGlobesData from './data/golden-globes.json'
-// import avocadoSalesData from './data/avocado-sales.json'
-// import booksData from './data/books.json'
-// import netflixData from './data/netflix-titles.json'
-// import topMusicData from './data/top-music.json'
-
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
-//   PORT=9000 npm start
 const port = process.env.PORT || 8080
 const app = express()
+const myEndPoints = require('express-list-endpoints')
+const ERROR_DATA_NOT_FOUND = { error: 'Data not found'}
 
-// Add middlewares to enable cors and json body parsing
 app.use(cors())
 app.use(bodyParser.json())
 
-// Start defining your routes here
 app.get('/', (req, res) => {
-  res.send('Hello world')
+  if (!res) {
+    res
+      .status(404)
+      .send({ error: 'Oops! Something goes wrong. Try again later' })
+  }
+  res.send(myEndPoints(app))
 })
 
+app.get('/books', (req, res) => {
+  const books = booksData
+  const page = req.query.page ? req.query.page -1 : 0 //page 0 as default, page number is -1 compared to the array index
+    const pageSize = req.query.pageSize ?? 20 //default is 20
+    const startIndex = page * pageSize
+    const endIndex = startIndex + pageSize
+    const bookPerPage = books.slice(startIndex, endIndex)
+    const returnObj = {
+      amountOfData: books.length,
+      amounfOfPages: Math.ceil(books.length / pageSize),
+      currentPage: page + 1,
+      pageSize: pageSize,
+      startIndex: startIndex,
+      endIndex: endIndex,
+      results: bookPerPage
+    }
```

Comment:
Nice and clean pagination code 👍🏻  Maybe it would be good to extract it to the separate function to gain the ability to reuse it?